master/buildbot/status/logfile.py

```diff
@@ -657,56 +657,28 @@ def __setstate__(self, d):
         self.finished = True
 
 
-class HTMLLogFile:
-    implements(interfaces.IStatusLog)
-
-    filename = None
+class HTMLLogFile(LogFile):
 
     def __init__(self, parent, name, logfilename, html):
-        self.step = parent
-        self.name = name
-        self.filename = logfilename
-        self.html = html
-
-    def getName(self):
-        return self.name  # set in BuildStepStatus.addLog
-
-    def getStep(self):
-        return self.step
-
-    def isFinished(self):
-        return True
-
-    def waitUntilFinished(self):
-        return defer.succeed(self)
+        LogFile.__init__(self, parent, name, logfilename)
+        self.addStderr(html)
+        self.finish()
 
     def hasContents(self):
+        assert not self._isNewStyle, "not available in new-style steps"
         return True
 
-    def getText(self):
-        return self.html  # looks kinda like text
-
-    def getTextWithHeaders(self):
-        return self.html
-
-    def getChunks(self):
-        return [(STDERR, self.html)]
-
-    def subscribe(self, receiver, catchup):
-        pass
-
-    def unsubscribe(self, receiver):
-        pass
-
-    def finish(self):
-        pass
+    def __setstate__(self, d):
+        self.__dict__ = d
+        self.watchers = []
+        self.finishedWatchers = []
+        self.finished = True
 
-    def __getstate__(self):
-        d = self.__dict__.copy()
-        del d['step']
-        if "master" in d:
-            del d['master']
-        return d
+        # buildbot <= 0.8.8 stored all html logs in the html property
+        if 'html' in self.__dict__:
+            buf = "%d:%d%s," % (len(self.html) + 1, STDERR, self.html)
+            self.openfile = StringIO(buf)
+            del self.__dict__['html']
```

Comment:
Looking back at this with @mariangemarcano I notice that this doesn't write the HTML-free pickle back to disk.  That might be a nice improvement for someone else (Maria?) to work on.